### PR TITLE
feat(PointerCursor): Support for WPF & GTK

### LINKED ISSUE #4078: Add support for PointerCursor on WPF and GTK
## What would you like to be added:

`PointerCursor` on WPF and GTK

## Why is this needed:

UWP parity

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4078 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported.

## What is the new behavior?

Supported.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by robloo** [comment]
I was just looking at this functionality. It appears there is nothing that can be done for macOS using officially documented APIs though.

**Comment by MartinZikmund** [comment]
> I was just looking at this functionality. It appears there is nothing that can be done for macOS using officially documented APIs though.
> 
> Edit: I spoke too soon. https://developer.apple.com/documentation/appkit/nscursor. Would you like me to take a look at adding this on macOS? I'm not sure if you plan on doing that yourself.

Actually it is implemented for macOS (at least for some of the pointer types, which have their counterparts 😀).

**Comment by robloo** [comment]
Haha, ok, I was getting warnings on the Android build and made a bad assumption. I should have checked the code!

File: src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenUIElementPointersSupport.cs [hunk 1]
```diff
@@ -33,13 +33,22 @@ public class TizenUIElementPointersSupport : ICoreWindowExtension
 		private readonly GestureLayer _gestureLayer;
 		private readonly UnoCanvas _canvas;
 
+		public CoreCursor PointerCursor
+		{
+			get => new CoreCursor(CoreCursorType.Arrow, 0);
+			set
+			{
+				// ignored
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenUIElementPointersSupport.cs:41]: Is it not supported or not implemented ?

File: src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+using Windows.UI.Core;
+
+namespace Uno.UI.Skia.Platform.Extensions
+{
+	internal static class CursorExtensions
+	{
+		public static CoreCursor ToCoreCursor(this Cursor cursor)
+		{
+			CoreCursorType cursorType = CoreCursorType.Arrow;
+			if (cursor == Cursors.Wait)
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs:11]: Feels like a good candidate for a switch expression ?

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenUIElementPointersSupport.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenUIElementPointersSupport.cs:41]: As far as I found, it is not supported. Maybe some Tizen devs could chime in here 😀

File: src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs:11]: I thought so, but the `Cursors.Xyz` are static properties. Is there a form of switch expression that can handle those elegantly?

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/Extensions/CursorExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+				type switch
+				{
+					CursorType.Arrow => CoreCursorType.Arrow,
+					CursorType.Cross => CoreCursorType.Cross,
+					CursorType.Hand1 => CoreCursorType.Hand,
+					CursorType.Hand2 => CoreCursorType.Hand,
+					CursorType.QuestionArrow => CoreCursorType.Help,
+					CursorType.Sizing => CoreCursorType.SizeAll,
+					CursorType.Watch => CoreCursorType.Wait,
+					_ => throw new NotImplementedException(),
```
Comment (carldebilly) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/Extensions/CursorExtensions.cs:25]: `NotSupportedException` may be more appropriate here.

**Review by carldebilly — APPROVED** [review]

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Wpf/WPF/Extensions/CursorExtensions.cs:11]: Right. Could be mixing up with C# 9.0 ;)

**Pull request merged**